src/Config/Parser/MetadataParser/MetadataParser.php

```diff
@@ -0,0 +1,956 @@
+
+        $interfaceConfiguration['fields'] = array_merge($fieldsFromProperties, $fieldsFromMethods);
+        $interfaceConfiguration = self::getDescriptionConfiguration(static::getMetadatas($reflectionClass)) + $interfaceConfiguration;
+
+        $interfaceConfiguration['resolveType'] = self::formatExpression($interfaceAnnotation->resolveType);
+
+        return ['type' => 'interface', 'config' => $interfaceConfiguration];
+    }
+
+    /**
+     * Create a GraphQL Input type configuration from metadatas on properties.
+     */
+    private static function inputMetadataToGQLConfiguration(ReflectionClass $reflectionClass, Metadata\Input $inputAnnotation): array
+    {
+        $inputConfiguration = array_merge([
+            'fields' => self::getGraphQLInputFieldsFromMetadatas($reflectionClass, self::getClassProperties($reflectionClass)),
+        ], self::getDescriptionConfiguration(static::getMetadatas($reflectionClass)));
+
+        return ['type' => $inputAnnotation->isRelay ? 'relay-mutation-input' : 'input-object', 'config' => $inputConfiguration];
+    }
+
+    /**
+     * Get a GraphQL scalar configuration from given scalar metadata.
+     */
+    private static function scalarMetadataToGQLConfiguration(ReflectionClass $reflectionClass, Metadata\Scalar $scalarAnnotation): array
+    {
+        $scalarConfiguration = [];
+
+        if (isset($scalarAnnotation->scalarType)) {
+            $scalarConfiguration['scalarType'] = self::formatExpression($scalarAnnotation->scalarType);
+        } else {
+            $scalarConfiguration = [
+                'serialize' => [$reflectionClass->getName(), 'serialize'],
+                'parseValue' => [$reflectionClass->getName(), 'parseValue'],
+                'parseLiteral' => [$reflectionClass->getName(), 'parseLiteral'],
+            ];
+        }
+
+        $scalarConfiguration = self::getDescriptionConfiguration(static::getMetadatas($reflectionClass)) + $scalarConfiguration;
+
+        return ['type' => 'custom-scalar', 'config' => $scalarConfiguration];
+    }
+
+    /**
+     * Get a GraphQL Enum configuration from given enum metadata.
+     */
+    private static function enumMetadataToGQLConfiguration(ReflectionClass $reflectionClass, Metadata\Enum $enumMetadata): array
+    {
+        $metadatas = static::getMetadatas($reflectionClass);
+        $enumValues = array_merge(self::getMetadataMatching($metadatas, Metadata\EnumValue::class), $enumMetadata->values);
+
+        $values = [];
+
+        foreach ($reflectionClass->getConstants() as $name => $value) {
+            $valueAnnotation = current(array_filter($enumValues, fn ($enumValueAnnotation) => $enumValueAnnotation->name === $name));
+            $valueConfig = [];
+            $valueConfig['value'] = $value;
+
+            if ($valueAnnotation && isset($valueAnnotation->description)) {
+                $valueConfig['description'] = $valueAnnotation->description;
+            }
```

Comment:
Note: when I see this code 
```php
if ($valueAnnotation && isset($valueAnnotation->description))
```
it's hard for me to understand, what did you mean with $valueAnnotation. Did you want to check if it's false? Null? An empty string? A null string "0"? An integer null? An empty array?

And in order to understand it, i need to analyse the code above:
```php
$valueAnnotation = current(array_filter($enumValues, fn ($enumValueAnnotation) => $enumValueAnnotation->name === $name));
```
Okay, I see it uses `array_filter`, then it's an array, then it uses `current`, which means I need to analyse, what values can `$enumValues` contain etc. This costs other developers times, plus loose comparisons increases the possibility of errors. 

That's another reason why I encourage you to use strict comparison, whenever possible and of course until it doesn't make the code unreadable 😉